extension/js/common/api/email-provider/gmail/gmail.ts

```diff
@@ -304,7 +305,7 @@ export class Gmail extends EmailProviderApi implements EmailProviderInterface {
       if (plaintext) {
         return { armored: '', plaintext, subject, isPwdMsg };
       }
-      throw new FormatError('Armored message not found', JSON.stringify(gmailMsg.payload, undefined, 2));
+      throw new FormatError('Armored message not found', base64decode(gmailMsg.payload!.parts![0].parts![0].body!.data as string));
```

Comment:
please explain the structure of `gmailMsg.payload`.
Why is it guaranteed to have `parts[0].parts[0]`?

Reply:
Hello @rrrooommmaaa,

1.) As far as I understand, no - I realize what if I wrap the throw formatError block in an if condition, for example:
```typescript
const bodies = GmailParser.findBodies(gmailMsg);
...
...
if (typeof bodies['text/plain'] === 'string') {
        throw new FormatError('Armored message not found', base64decode(bodies['text/plain']));
}
```
or something similar to ensure that every property has a value that is not undefined.


Another problem is that - I'm unable to reproduce the issue even on an older version 8.2.4. However, I'm worrying that leaving this JSON.stringify in the code would not be good? 

2.) I have prepared the test message within the compatibility test account. It has the subject name of `Bad Format produces json output`. However, I'm no longer able to replicate it now - Tbh, there are some users that are experiencing this but reinstalling the extension helps them resolve the issue. 

Here it is from the email (support): https://mail.google.com/mail/u/6/#inbox/FMfcgzGpFgmLZvDsTLCKhgRRDVLxhWhQ

Reply:
The error pop up once again:

I'm using firefox browser 99.0.1 in ubuntu and a flowcrypt extension copy from flowcrypt.com and updated to 8.2.8. I'm not experiencing the error when using firefox on a mac. this is really strage.

Reply:
We already have `textBody` and `htmlBody` variables, why not reuse them?
`throw new FormatError('Armored message not found', textBody || htmlBody);`

Reply:
I applied the changes - thanks for your patience. I will try to be more attentive next time.